src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/DeclaratorHandler.java

```diff
@@ -89,14 +90,36 @@ private Declaration handleDeclarator(CPPASTDeclarator ctx) {
       return handle(ctx.getNestedDeclarator());
     }
 
-    // type will be filled out later
-    VariableDeclaration declaration =
+    Declaration declaration;
+
+    declaration =
         NodeBuilder.newVariableDeclaration(
             ctx.getName().toString(), UnknownType.getUnknownType(), ctx.getRawSignature(), true);
     IASTInitializer init = ctx.getInitializer();
-
     if (init != null) {
-      declaration.setInitializer(lang.getInitializerHandler().handle(init));
+      ((VariableDeclaration) declaration).setInitializer(lang.getInitializerHandler().handle(init));
+    }
+    String name = ctx.getName().toString();
+    if (name.contains(lang.getNamespaceDelimiter())) {
+
+      String[] rr = name.split(lang.getNamespaceDelimiter());
+
+      String recordName =
+          String.join(lang.getNamespaceDelimiter(), Arrays.asList(rr).subList(0, rr.length - 1));
+      String fieldName = rr[rr.length - 1];
+
+      RecordDeclaration recordDeclaration = this.lang.getRecordForName(recordName).orElse(null);
+
+      declaration = FieldDeclaration.from((VariableDeclaration) declaration);
+      if (recordDeclaration != null) {
+        Optional<FieldDeclaration> existingDeclaration =
+            recordDeclaration.getFields().stream()
+                .filter(f -> f.getName().equals(fieldName))
+                .findFirst();
+        if (existingDeclaration.isPresent()) {
+          existingDeclaration.get().setDefinition((FieldDeclaration) declaration);
```

Comment:
As mentioned in the comments, can you provide an example for this? I am still not sure if field declarations can have a definition.